packages/tamagotchi/src/main/index.ts

```diff
@@ -0,0 +1,100 @@
+import { readFileSync } from 'node:fs'
+import { dirname, join } from 'node:path'
+import { platform } from 'node:process'
+import { fileURLToPath } from 'node:url'
+import { electronApp, is, optimizer } from '@electron-toolkit/utils'
+import { app, BrowserWindow, ipcMain, shell } from 'electron'
+import icon from '../../build/icon.png?asset'
+
+const __dirname = dirname(fileURLToPath(import.meta.url))
+
+function createWindow(): void {
+  // Create the browser window.
+  const mainWindow = new BrowserWindow({
+    width: 300,
+    height: 400,
+    show: false,
+    autoHideMenuBar: true,
+    frame: false,
+    transparent: true,
+    resizable: false,
+    hasShadow: false,
+    alwaysOnTop: true,
+    ...(platform === 'linux' ? { icon } : {}),
+    webPreferences: {
+      preload: join(__dirname, '..', 'preload', 'index.js'),
```

Comment:
```suggestion
      preload: join(import.meta.dirname, '..', 'preload', 'index.js'),
```

No more `__dirname`, just use [`import.meta.dirname`](https://nodejs.org/docs/latest/api/esm.html#importmeta).